src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/Controller.cs

```diff
@@ -154,6 +99,67 @@ public QuickInfoService GetService()
                 throw ExceptionUtilities.Unreachable;
             }
         }
+
+        public QuickInfoService GetService()
+        {
+            if (_service == null)
+            {
+                var snapshot = this.SubjectBuffer.CurrentSnapshot;
+                var document = snapshot.GetOpenDocumentInCurrentContextWithChanges();
+                if (document != null)
+                {
+                    _service = QuickInfoService.GetService(document);
+                }
+            }
+
+            return _service;
+        }
+
+        private IntellisenseQuickInfoItem BuildIntellisenseQuickInfoItem(SnapshotPoint triggerPoint,
+            ITextSnapshot snapshot, CodeAnalysisQuickInfoItem quickInfoItem)
+        {
+            var line = triggerPoint.GetContainingLine();
+            var lineSpan = snapshot.CreateTrackingSpan(
+                line.Extent,
+                SpanTrackingMode.EdgeInclusive);
+
+            // Build the first line of QuickInfo item, the images and the Description section should be on the first line with Wrapped style
+            var glyphs = quickInfoItem.Tags.GetGlyphs();
+            var symbolGlyph = glyphs.FirstOrDefault(g => g != Glyph.CompletionWarning);
+            var warningGlyph = glyphs.FirstOrDefault(g => g == Glyph.CompletionWarning);
+            var firstLineElements = new List<Object>();
+            if (symbolGlyph != Glyph.None)
+            {
+                firstLineElements.Add(new ImageElement(symbolGlyph.GetImageId()));
+            }
+
+            if (warningGlyph != Glyph.None)
+            {
+                firstLineElements.Add(new ImageElement(warningGlyph.GetImageId()));
+            }
+
+            firstLineElements.Add(BuildClassifiedTextElement(
+                quickInfoItem.Sections.Where(s => s.Kind == QuickInfoSectionKinds.Description).FirstOrDefault()));
+
+            var elements = new List<object>();
+            elements.Add(new ContainerElement(ContainerElementStyle.Wrapped, firstLineElements));
+
+            // Add the remaining sections as Stacked style
+            elements.AddRange(
+                quickInfoItem.Sections.Where(s => s.Kind != QuickInfoSectionKinds.Description)
+                                      .Select(section => BuildClassifiedTextElement(section)));
+
+            var content = new ContainerElement(
+                                ContainerElementStyle.Stacked,
+                                elements);
+
+            return new IntellisenseQuickInfoItem(lineSpan, content);
```

Comment:
Reconsider the applicableToSpan. I know that my API sample used the span of the entire line, but this is probably not what you want, because doing so will prevent the quick info from being dismissed as you move the mouse from one token on the line to another.

You probably want to return just the span for the token under the mouse.

Reply:
almost missed this one. fixed it now. Thanks! :)